pkg/core/xds/types.go

```diff
@@ -177,7 +177,7 @@ type ExternalServiceDynamicPolicies map[ServiceName]PluginOriginatedPolicies
 
 type MeshResources struct {
 	Mesh                           *core_mesh.MeshResource
-	TrafficRoutes                  []*core_mesh.TrafficRouteResource
+	Resources                      map[core_model.ResourceType]core_model.ResourceList
```

Comment:
For new plugin-originated policy types, we already have a field `MeshResources.Dynamic`. So I'd keep this field as it is and took MeshHTTPRoute policies from the `Dynamic`.

Reply:
You have to implement `EgressMatchedPolicies` method on MeshHTTPRoute plugin so the policies are present in `Dynamic`

Reply:
Right, `Dynamic` is for ExternalServices. In that case, I'd either rename `Resources` to `Routes` or simply create the second field:
```go
TrafficRoutes  []*core_mesh.TrafficRouteResource
MeshHTTPRoutes []*MeshHTTPRouteResource
```

Reply:
Added second field